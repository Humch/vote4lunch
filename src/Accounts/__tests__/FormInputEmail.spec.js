/* eslint-disable no-undef */
import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { FormInputEmail } from '..';
import configureStore from 'redux-mock-store';

describe('FormInputEmail Snapshot', () => {
  const initialState = { voteDataForm: { email: 'bob@bob.com' } };
  const mockStore = configureStore();
  let store;

  beforeEach(() => {
    store = mockStore(initialState);
  });

  it('capturing Snapshot of FormInputEmail', () => {
    const renderedValue = renderer.create(<FormInputEmail store={store} />).toJSON();
    expect(renderedValue).toMatchSnapshot();
  });
});

describe('FormInputEmail - Shallow Render REACT COMPONENTS', () => {
  const initialState = { voteDataForm: { email: 'bob@bob.com' } };
  const mockStore = configureStore();
  let store;
  let wrapper;

  beforeEach(() => {
    store = mockStore(initialState);
    wrapper = shallow(<FormInputEmail store={store} />);
  });

  it('render the DUMB component', () => {
    expect(wrapper.length).toEqual(1);
  });

  it('label value', () => {
    expect(wrapper.find('label').get(0).props.children).toBe('Indique ton email');
  });

  it('contains email', () => {
    expect(wrapper.find('input').prop('value')).toEqual(voteDataForm.email);
});
});

describe('FormInputEmail (Shallow + passing the {store} directly)', () => {
  const initialState = { voteDataForm: { email: 'bob@bob.com' } };
  const mockStore = configureStore();
  let store;
  let container;

  beforeEach(() => {
    store = mockStore(initialState);
    container = shallow(<FormInputEmail store={store} />);
  });

  it('render the connected(SMART) component', () => {
    expect(container.length).toEqual(1);
  });

  it('check Prop matches with initialState', () => {
    expect(container.prop('email')).toEqual(initialState.voteDataForm.email);
  });
});
