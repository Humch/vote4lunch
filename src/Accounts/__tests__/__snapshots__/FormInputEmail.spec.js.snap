// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormInputEmail Snapshot capturing Snapshot of FormInputEmail 1`] = `
<div
  className="form-group"
>
  <label
    className="text-white"
    htmlFor="email"
  >
    Indique ton email
  </label>
  <input
    className="form-control"
    id="email"
    name="email"
    onChange={[Function]}
    required={true}
    type="email"
    value="bob@bob.com"
  />
</div>
`;
